src/main/java/uk/ac/cam/cl/dtg/segue/api/AuthorisationFacade.java

```diff
@@ -104,20 +104,31 @@ public AuthorisationFacade(final PropertiesLoader properties, final UserAccountM
     }
 
     /**
-     * Get all users who can see my data.
+     * Get all users who can see the specified users data.
      * 
      * @param request
      *            - so we can identify the current user.
+     * @param userId
+     *            - the user id we want the associations of.
+     * @throws NoUserException
+     *            - when the target user cannot be found.
      * @return List of user associations.
      */
     @GET
-    @Path("/")
+    @Path("/{userId}")
```

Comment:
Hang on, are we OK with losing an endpoint like this? @jsharkey13 @mlt47 @sacummins

Reply:
They could be combined with `"/{userId: .*}"` and then have the id fallback to that of the current user.

Reply:
Yes, if that's an option, it also allows us to keep the code compact. Go for it.

Reply:
It appears @Path only supports paths which still have at least one character excluding the trailing slash e.g. `authorisations/foo/{userId: .*
}` matches with `authorisations/foo` and `/foo/123` but `{userId: .*}` doesn't match `authorisations/` since the part in @Path only contains a slash.
Instead should I separate the endpoints?

Reply:
I think these should be separate endpoints. But it is likely that you can make the shortcut one call the Java function for the specified user one, passing in the Request object and setting the current user's ID as the target.

Your existing permission check should cover that use case just fine.